sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -123,7 +123,10 @@ export class StreamingReceiver extends MessageReceiver {
    * @param {ReceiveOptions} [options]     Receive options.
    * @return {StreamingReceiver} An instance of StreamingReceiver.
    */
-  static create(context: ClientEntityContext, options?: ReceiveOptions): StreamingReceiver {
+  static async create(
+    context: ClientEntityContext,
+    options?: ReceiveOptions
+  ): Promise<StreamingReceiver> {
```

Comment:
Since you reverted the change which moved the _init() inside create(), this no longer has to be an async function

Reply:
The messageHandler is attached only when the sReceiver is setup and returned, hence the async / promise changes are required?

Reply:
This here is forcing a synchronous function to be async.
The message handler (I am assuming you are referring to the `onAmqpMessage` in the parent class `messageReceiver`) is attached as soon as the underlying rhea receiver is created inside the `_init` function.

Reply:
Moved _init() to create. Refer to https://github.com/Azure/azure-sdk-for-js/pull/2139#issuecomment-482215941